mitosheet/src/components/endo/FormulaBar.tsx

```diff
@@ -5,24 +5,46 @@ import React from 'react';
 // Import css
 import "../../../css/FormulaBar.css";
 import "../../../css/mito.css"
-import { EditorState, SheetData, MitoSelection } from '../../types';
-import { getFullFormula } from './cellEditorUtils';
+import { EditorState, SheetData, MitoSelection, GridState, UIState } from '../../types';
+import { getFullFormula } from './celleditor/cellEditorUtils';
 import { getCellDataFromCellIndexes } from './utils';
 import Col from '../spacing/Col';
 import Row from '../spacing/Row';
+import MitoAPI from '../../jupyter/api';
+import { calculateCurrentSheetView } from './sheetViewUtils';
+import CellEditor from './celleditor/CellEditor';
 
 const FormulaBar = (props: {
-    sheetData: SheetData | undefined,
+    sheetData: SheetData,
+    sheetIndex: number,
+    gridState: GridState,
+    editorState: EditorState | undefined,
+    setEditorState: React.Dispatch<React.SetStateAction<EditorState | undefined>>,
+    setGridState: React.Dispatch<React.SetStateAction<GridState>>,
+    setUIState: React.Dispatch<React.SetStateAction<UIState>>,
+    scrollAndRenderedContainerRef: React.RefObject<HTMLDivElement>,
+    containerRef: React.RefObject<HTMLDivElement>,
+    mitoAPI: MitoAPI,
     selection: MitoSelection,
-    editorState: EditorState | undefined;
 }): JSX.Element => {
 
-    const {columnHeader, columnFormula, cellValue} = getCellDataFromCellIndexes(props.sheetData, props.selection.endingRowIndex, props.selection.endingColumnIndex);
+    const rowIndex = props.selection.startingRowIndex
+    const colIndex = props.selection.startingColumnIndex
+
+    const {columnHeader, columnFormula, cellValue} = getCellDataFromCellIndexes(props.sheetData, rowIndex, colIndex);
     const originalFormulaBarValue = '' + (columnFormula !== undefined && columnFormula !== '' ? columnFormula : (cellValue !== undefined ? cellValue : ''));
     const cellEditingCellData = props.editorState === undefined ? undefined : getCellDataFromCellIndexes(props.sheetData, props.editorState.rowIndex, props.editorState.columnIndex);
     const formulaBarColumnHeader = props.editorState === undefined ? columnHeader : cellEditingCellData?.columnHeader;
-    const formulaBarValue = props.editorState === undefined ? originalFormulaBarValue : getFullFormula(props.editorState.formula, formulaBarColumnHeader || '', props.editorState.pendingSelectedColumns);
 
+    // If the formula bar is a cell, display the cell value. If it is a column header, display the column header
+    let formulaBarValue = '' 
+    if (rowIndex == -1 && columnHeader !== undefined) {
+        formulaBarValue = String(columnHeader)
```

Comment:
I don't think this is correct to cast to a string like this. We have a function called `getDisplayColumnHeader` - which I think is what you want...

The fact that we _never_ use the `String(` utility on column headers before (but we do have column headers displayed as text all over the place) is good evidence that this approach is not what we want. 

You've gotten got on type casting before. I think when you find yourself converting between types, try and find a similar example to help guide you!